### PR TITLE
feat: introduce modexp precompile

### LINKED ISSUE #1456: MODEXP precompile

### PR DESCRIPTION
**Pull request by letypequividelespoubelles**
> [!NOTE]
> **Medium Risk**
> Adds new arithmetic primitives and recursive exponentiation logic in zkasm, which can be correctness- and performance-sensitive; changes are mostly additive and covered by new acceptance tests but may still have edge-case risk.
> 
> **Overview**
> Introduces a new zkasm implementation of MODEXP for 256-bit inputs via `modexp_u256`, using a recursive square-and-multiply approach with early exits for trivial modulus/exponent/base cases, and adds a corresponding `.accepts` test vector set (including an EIP-198 case).
> 
> Adds a reusable `modulus_u512_u256` helper and refactors/extends the bench `ext` module (ADDMOD/MULMOD) to compute modulo via this helper; also adds an initial `modexp_u8192` implementation and makes small fixes/updates to MOD bench artifacts (typo fix and a couple of new/adjusted accepts).
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit afe2481b05b87acff5b1e84d0b38c210632f32b1. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: testdata/asm/util/modulus.zkasm [hunk 1]
```diff
@@ -0,0 +1,17 @@
+;; compute a mod b where a is a u512 and a u256
+;; a mod 0 = 0
+fn modulus_u512_u256(a u512, b u256) -> (res u256) {
+  var quot u512
+  var sum u768
+  var witsum u769
```
Comment (DavePearce) [inline testdata/asm/util/modulus.zkasm]: Your `witsum` only needs to be `u257` in this case.

File: testdata/asm/util/modexp_u256.zkasm [hunk 2]
```diff
@@ -0,0 +1,47 @@
+;; MODEXP precompile where all input are at most 256 bits long (ie are 32 bytes ie EWORD)
+
+include "../bench/mod.zkasm"
+include "modulus.zkasm"
+
+pub fn modexp_u256(base u256, exponent u256, modulus=1 u256) -> (result u256){
+
+;; some trivial output
```
Comment (DavePearce) [inline testdata/asm/util/modexp_u256.zkasm]: Generally in assembly language we indent everything except labels
